Assignment4/Assignment4.py

```diff
@@ -0,0 +1,246 @@
+# importing necessary libraries
+import sys
+import pandas as pd
+import numpy
+from plotly import express as px
+from plotly import figure_factory as ff
+from plotly import graph_objects as go
+from sklearn.metrics import confusion_matrix
+import plotly.express as px
+import statsmodels.api
+from sklearn.ensemble import RandomForestRegressor
+
+# Using the different data to test the code
+
+def Cont_resp():
+    # loading the data
+    data = pd.read_csv("/content/drive/MyDrive/auto-mpg.csv")
+
+    # Separating the data to target and response variables
+    X = data[["cylinders", "displacement", "weight", "acceleration", "model year", "origin"]]  # PREDICTOR
+    Y = data[["mpg"]]  # RESPONSE
+
+    return data, X, Y
+
+
+def Cat_resp():
+    # loading the data
+    data = pd.read_csv("/content/drive/MyDrive/heart_failure_clinical_records_dataset.csv")
+
+    # Separating the data to target and response variables
+    X = data[
+        ["creatinine_phosphokinase", "diabetes", "ejection_fraction", "high_blood_pressure", "platelets", "platelets",
+         "sex", "smoking", "time"]]  # PREDICTOR
+    Y = data[["DEATH_EVENT"]]  # RESPONSE
+
+    return data, X, Y
+
+
+def Cat_resp2():
```

Comment:
You should create functions to check type of response and predictors as you will check multiple times.